### DISCUSSION TITLE
passing timeline variable to loop function

### DISCUSSION THREAD

**Discussion by mllewis** [jspsych/jsPsych#1050] [Q&A]
Each trial in my experiment needs to be repeated n times where n varies by trial, and is defined in a timeline variable.

The solution I've come up with is to use a loop_function, but I am unable to pass information from the original timeline variable to the loop function. Can this be done?

This is what the relevant bit of code looks like:

```javascript
 var loop_activities_description_responses = {
    timeline: [activities_responses, description_responses],
    loop_function: function(current_char_num){
        current_char_num += 1
        if(jsPsych.timelineVariable('char_name', true)[current_char_num] == null){
            current_char_num = 0
            return true;
        } else {
            return false;
        }
    }
  };

  var test_procedure = {
    timeline: [show_text, loop_activities_description_responses],
    timeline_variables: book_stimuli_shuffled
  }
```

And, this is what one element in book_stimuli_shuffled looks like:

`{"book_id":"L111","title":"TRIANGLE (SHAPE TRILOGY)","text":"blah","char_name":["Miles","mama"]}
`

Thank you!

**Comment by jodeleeuw** [comment]
Hi @mllewis ,

Yes, you can do this. But the `loop_function` argument is always the data generated by the latest iteration of the timeline. I think what you need to do is move `current_char_num` to a global variable, and drop it from the function parameters list.

```js

var current_char_num = 0;

 var loop_activities_description_responses = {
    timeline: [activities_responses, description_responses],
    loop_function: function(){
        current_char_num += 1
        if(jsPsych.timelineVariable('char_name', true)[current_char_num] == null){
            current_char_num = 0
            return true;
        } else {
            return false;
        }
    }
  };

  var test_procedure = {
    timeline: [show_text, loop_activities_description_responses],
    timeline_variables: book_stimuli_shuffled
  }
```
